crypto/x509/by_store.c

```diff
@@ -91,21 +108,24 @@ static int cache_objects(X509_LOOKUP *lctx, const char *uri,
             break;
     }
     OSSL_STORE_close(ctx);
+    store->ctx = NULL;
 
     return ok;
 }
 
 
-/* Because OPENSSL_free is a macro and for C type match */
-static void free_uri(OPENSSL_STRING data)
+static void free_store(CACHED_STORE *store)
 {
-    OPENSSL_free(data);
+    OSSL_STORE_close(store->ctx);
```

Comment:
Please add `if (store == NULL) return;` to make it similar to other free functions.

Reply:
Fixed